### DISCUSSION TITLE
Mimicking IntelliJ's sorting of imports

### DISCUSSION THREAD

**Discussion by Overhear8805** [mfussenegger/nvim-jdtls#488] [Q&A]
![image](https://github.com/mfussenegger/nvim-jdtls/assets/3425747/15a6a5e8-efd7-4394-843e-36b518c4c87b)
_Example of a file IntelliJ has organized imports of_

![image](https://github.com/mfussenegger/nvim-jdtls/assets/3425747/71391d36-8986-4567-9155-49fb3d9b4b8f)
_Example of a file jdtls has organized imports of_

I'm working in a team of IntelliJ users and IntelliJ has a different import sorting order than jdtls, resulting in unnecessary code changes every time I run "Organize imports". 

Is there a way to configure nvim-jdtls to organize the imports in the same order as IntelliJ?

**Comment by gbroques** [comment]
Hi @simon1573.

We had a similar problem at my workplace with Visual Studio Code & IntelliJ users.

You can change the import order using the `java.completion.importOrder` setting.

For `nvim-jdtls`, you can define this in Lua like:
```lua
config.settings = {
  java = {
    completion = {
      importOrder = {
        "java",
        "javax",
        "org",
        "com"
      }
    }
  }
}
````
See @mfussenegger's sample configuration for more context:
https://github.com/mfussenegger/dotfiles/blob/833d634251ebf3bf7e9899ed06ac710735d392da/vim/.config/nvim/ftplugin/java.lua#L12-L27

See https://github.com/redhat-developer/vscode-java#supported-vs-code-settings for more details on this configuration option, and others that are available.

There's also static imports to consider.

I believe JDTLS will layout static imports separately with no way to change this behavior, but this is an option in IntelliJ.

See [Intellij Import Layout options](https://www.jetbrains.com/help/idea/code-style-java.html#imports_table):
> **Layout static imports separately:** if this checkbox is selected, all static imports will be kept in a separate section. Otherwise, all import statements will be sorted according to the specified layout rules.

I believe IntelliJ has a similar ability to specify the import order somewhere.

I would reach out to your IntelliJ co-workers to determine what IntelliJ's default import order is, or have everyone define the same order.
